File: frontend/src/models/Feeding.ts
export class FeedingDto {

  id: number;
  catname: string;
  brandname: string;
  productname: string;
  taste: string;
  type: string;
  eatenpercentage: number;
  created: string;
  feedingtime: string;

  constructor(data: any){
    data = data || {};
    this.id = data.id;
    this.catname = data.catname;
    this.brandname = data.brandname;
    this.productname = data.productname;
    this.taste = data.taste;
    this.type = data.type;
    this.eatenpercentage = data.eatenpercentage;
    this.created = data.created;
    this.feedingtime = data.feedingtime;
  }

  feedingTimeGerman(): string {
    const date = new Date(this.feedingtime);
    return date.toLocaleString();
  }

}

File: backend/Models/Feeding.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace backend.Models
{
  public class Feeding
  {
    public int id { get; set; }
    
    [Required]
    public string catname { get; set; }
    
    [Required]
    public string brandname { get; set; }
    
    [Required]
    public string productname { get; set; }    
    
    [Required]
    public string taste { get; set; }    
    
    [Required]
    public string type { get; set; }    
    
    [Required]
    public int eatenpercentage { get; set; }
    public DateTime? created { get; set; }
    
    [Required]
    public DateTime? feedingtime { get; set; }

    // Ab hier z.B. relationships (navigation properties) im Constructor initialisieren
    
  }
}